server/src/main/java/org/elasticsearch/index/search/SimpleQueryStringQueryParser.java

```diff
@@ -90,7 +91,7 @@ private Analyzer getAnalyzer(MappedFieldType ft) {
      */
     private Query rethrowUnlessLenient(RuntimeException e) {
         if (settings.lenient()) {
-            return null;
+            return Queries.newMatchNoDocsQuery("failed query, caused by " + e.getMessage());
```

Comment:
nit: can you update the comment to refect that we return MatchNoDocs instead of null now?